src/Tools/ExternalAccess/Razor/IRazorAsynchronousOperationListenerProviderAccessor.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    internal interface IRazorAsynchronousOperationListenerProviderAccessor
+    {
+        IAsynchronousOperationListener GetListener(string featureName);
```

Comment:
```suggestion
        RazorAsynchronousOperationListenerWrapper GetListener(string featureName);
```

Reply:
Here's another wrapper:

https://github.com/dotnet/roslyn/blob/master/src/Tools/ExternalAccess/Razor/Remote/RazorRemoteServiceCallbackIdWrapper.cs

You won't need the data contract attributes in this case, but the wrapper concept would be the same.